### PR TITLE
[TASK] Add section about the ``DefaultPage`` prototype approach

### PR DESCRIPTION
**Pull request by gerhard-boden**
source: https://discuss.neos.io/t/document-node-type-best-practices/242
The credit for this approach goes to Christopher Hlubek.

### CODE REVIEW

File: TYPO3.Neos/Documentation/HowTos/SelectingPageTemplate.rst [hunk 1]
```diff
@@ -89,13 +89,85 @@ the `page` path by default. But you can add your own conditions to render a diff
...
 This will now render the `employeePage` TypoScript path if a page of type `VendorName.VendorSite:Employee`
-is rendered on your website.
\ No newline at end of file
+is rendered on your website.
+
+Using a `DefaultPage` Prototype
+===============================
+
+This is an alternative and more flexible approach to the `Select Template based on NodeType` method descriped above.
+First we adjust the `default` `root` matcher not to render the page path, but a prototype derived from the current document node type name instead::
```
Comment (dimaip) [inline TYPO3.Neos/Documentation/HowTos/SelectingPageTemplate.rst]: Maybe write "`/page` path"?

File: TYPO3.Neos/Documentation/HowTos/SelectingPageTemplate.rst [hunk 2]
```diff
@@ -89,13 +89,85 @@ the `page` path by default. But you can add your own conditions to render a diff
...
+        # custom properties for your node type
+    }
+
+In case we have a `layout` property within our node type configuration, we can define a prototype for this case too::
+
+    customLayout = TYPO3.Neos.NodeTypes:Page.Document {
+       # custom properties for your node type
+    }
+
+The credit for this approach goes to Christopher Hlubek.
```
Comment (dimaip) [inline TYPO3.Neos/Documentation/HowTos/SelectingPageTemplate.rst]: I don't know if we should add credits to docs, looks a bit weird for official docs, wdyt?
Maybe just mentioning @hlubek in a commit message would be enough?

**Comment by dimaip** [comment]
I would really like to see this method as default rendering method in Neos, if we could do it in a not so breaking way.

**Comment by hlubek** [comment]
Hi there, I'm okay if it's just in the commit. No need for credits in the docs ;) I'm happy if we can push this approach as the default to Neos.

I played around with how to do this in a non-breaking manner, but this needs more research. I guess with a few TypoScript helpers for conditionally checking if a prototype is defined, we could implement a fallback to the old rendering.

**Comment by dimaip** [comment]
@hlubek Maybe we can at least add a new case to root object, and make old behavior disableable via TypoScript, then on next major release we can make it a breaking change, with which you could get the old behavior back also just with one line of TS.

**Comment by gerhard-boden** [comment]
I could resolve my problem from 2 days ago, luckily it didn't take too long since the error message actually implied what was wrong (who would have thought) :smiley: 
I found out that with this new approch you have to add the full package path to your node name, when using ``TYPO3.Neos:Content` as a supertype.
Before i didn't not have to write ``Vendor.SitePackageName: Nodename``, ``Vendor: Nodename`` used to be enough. I'll mention that in the documentation. Can someone else confirm this?

**Comment by kitsunet** [comment]
Do we have an agreement to merge this? :+1: from my side.

**Comment by skurfuerst** [comment]
:+1: for merging this!! Merging with previous votes.

**Pull request merged**